src/rtc/rtc_s1_single_job.py

```diff
@@ -270,14 +320,45 @@ def append_metadata_to_geotiff_file(input_file, metadata_dict):
            Input GeoTIFF file
        metadata_dict : dict
            Metadata dictionary
+       product_id : str
+           Product ID
     '''
     input_file_basename = os.path.basename(input_file)
     logger.info('    appending metadata to the GeoTIFF file:'
                 f' {input_file_basename}')
     gdal_ds = gdal.Open(input_file, gdal.GA_Update)
     existing_metadata = gdal_ds.GetMetadata()
+
+    # Update existing metadata with RTC-S1 metadata
     existing_metadata.update(metadata_dict)
+    layer_id = input_file_basename.replace(f'{product_id}_', '').split('.')[0]
+
+    # Update metadata file name
+    existing_metadata['FILE_NAME'] = input_file_basename
+
+    # Update metadata layer name (short description)
+    if layer_id in layer_names_dict.keys():
+        layer_name = layer_names_dict[layer_id]
+        existing_metadata['LAYER_NAME'] = layer_name
+
+        # Save layer name using SetDescription()
+        gdal_ds.SetDescription(layer_name)
+
+        band_out = gdal_ds.GetRasterBand(1)
+        band_out.SetDescription(layer_name)
+        band_out.FlushCache()
+
+        del band_out
+
+    # Update metadata layer description (long description)
+    if layer_id in layer_description_dict.keys():
+        layer_description = layer_description_dict[layer_id]
+        existing_metadata['LAYER_DESCRIPTION'] = layer_description
+
+    # Write metadata
     gdal_ds.SetMetadata(existing_metadata)
+
+    # Close GDAL dataset
     del gdal_ds
 
 
```

Comment:
In the function `_separate_pol_channels()` below, please check if the parameter `pol_list`, and `output_radiometry_str` are used in the function.